src/plugins/discover/public/application/main/services/load_saved_search.ts

```diff
@@ -68,7 +68,7 @@ export const loadSavedSearch = async (
 
   // Update saved search by a given app state (in URL)
   if (appState) {
-    if (savedSearchId && appState.index) {
+    if (savedSearchId && appState.index && !nextSavedSearch.isTextBasedQuery) {
```

Comment:
this is a change unrelated to this PR, it fixes an issue that the URL state is overwriting the data view index id of an text based index pattern, it should be taken care seperately.